src/back/spv/block.rs

```diff
@@ -1233,17 +1233,32 @@ impl<'w> BlockContext<'w> {
         Ok(pointer)
     }
 
-    fn get_expression_global(&self, expr_handle: Handle<crate::Expression>) -> Word {
-        match self.ir_function.expressions[expr_handle] {
+    fn get_image_id(
+        &mut self,
+        block: &mut Block,
+        expr_handle: Handle<crate::Expression>,
+    ) -> Result<Word, Error> {
+        Ok(match self.ir_function.expressions[expr_handle] {
             crate::Expression::GlobalVariable(handle) => {
                 let id = self.writer.global_variables[handle.index()].handle_id;
                 if id == 0 {
                     unreachable!("Global variable {:?} doesn't have a handle ID", handle);
                 }
                 id
             }
+            crate::Expression::FunctionArgument(i) => {
+                let result_type_id = self.get_type_id(LookupType::Handle(
+                    self.ir_function.arguments[i as usize].ty,
+                ))?;
+                let pointer_id = self.function.parameter_id(i);
+                let id = self.gen_id();
+                block
+                    .body
+                    .push(Instruction::load(result_type_id, id, pointer_id, None));
```

Comment:
this seems strange, a bit. Is `global_variables` storing the loaded value in the other branch? There is no reason why they should diverge. If the other doesn't load, this one shouldn't either.

Reply:
The globals are implicitly loaded in `writer.rs:407`, I could add a new struct similar to `GlobalVariable` (or reuse it).

Reply:
I think my general point still stands - this code needs to be more straightforward. As such, we can either emit the `OpLoad` expressions on all of the arguments of the function that are handles at the beginning of the function. Or we can stop trying to do this for globals as well ahead of time, and instead only `OpLoad` at the time of need, e.g. right before the sampling instruction.

Reply:
I've pushed some changes that make the global be loaded as needed let me know what you think about it